feat: Add Hungary chickenpox dataset

sabasiddique1 opened the pull request:
Refs  #2459

## List of changes to the codebase in this pull request
- Added Hungary Chickenpox dataset class (_hungary_chickenpox.py)
- 20 variables, 522 samples
- Continuous data type
- No expert knowledge
- Updated __init__.py to import HungaryChickenpox
- Added hungary_chickenpox to test file ALL_DATASETS list

## Description
What was added:

- New dataset file: pgmpy/datasets/_hungary_chickenpox.py
- Dataset registration in __init__.py
- Test coverage in test_datasets.py

sabasiddique1 committed:
[ENH] Add Hungary chickenpox dataset

sabasiddique1 committed:
[ENH] Add space in filie _hungary_chickenpox

sabasiddique1 committed:
Merge branch 'dev' into feature/2459/hungary-chickenpox

sabasiddique1 committed:
Merge branch 'dev' into feature/2459/hungary-chickenpox

sabasiddique1 committed:
Merge branch 'dev' into feature/2459/hungary-chickenpox

sabasiddique1 committed:
Merge branch 'dev' into feature/2459/hungary-chickenpox

sabasiddique1 committed:
Update __init__.py: add HungaryChickenpox dataset import

sabasiddique1 committed:
Merge branch 'feature/2459/hungary-chickenpox' of github.com:sabasiddique1/pgmpy into feature/2459/hungary-chickenpox

ankurankan committed:
Merge branch 'dev' into feature/2459/hungary-chickenpox

ankurankan reviewed (changes requested):
@sabasiddique1 This does seem to have ground truth data: https://github.com/pgmpy/example-causal-datasets/blob/main/real/hungary-chickenpox/ground.truth/hungary_county_edges.txt

ankurankan commented:
@sabasiddique1 BTW, for ground trut,h we rewrite the ground truth graph in dagitty syntax. Have a look at the Sachs dataset for example.

ankurankan committed:
Merge branch 'dev' into feature/2459/hungary-chickenpox

ankurankan committed:
Update and rename _hungary_chickenpox.py to hungary_chickenpox.py

ankurankan committed:
Update hungary_chickenpox.py

ankurankan commented:
Expert knowledge file still needs to be added.

ankurankan committed:
Merge branch 'dev' into feature/2459/hungary-chickenpox

ankurankan committed:
Update dataset metadata for Hungary chickenpox

ankurankan committed:
Adds a has_index_col tag to all datasets (#2569)

### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this
repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix
branch** (right side) against our **dev branch** (left side).
- [x] Are all the GitHub Actions checks passing? If not, you will need
to make changes to fix them. You can reference actions logs for the
failing check to identify the issue.

### Issue number(s) that this pull request fixes
- Fixes #2492

ankurankan committed:
Updates

ankurankan committed:
Fixes hungary chickenpox

ankurankan committed:
Remove hungary chickenpox from tests
